src/axom/quest/tests/quest_initialize.cpp

```diff
@@ -72,6 +75,76 @@ TEST(quest_initialize, signed_distance_pointer_initialize)
   delete input_mesh;
 }
 
+// Test allocation/reallocation using axom::allocate and axom::reallocate bytes
+TEST(quest_initialize, byte_reallocations)
+{
+  std::int8_t* b1 = axom::allocate<std::int8_t>(40);
+  EXPECT_NE(b1, nullptr);
+
+  b1 = axom::reallocate<std::int8_t>(b1, 80);
+  EXPECT_NE(b1, nullptr);
+
+  axom::deallocate(b1);
+}
+
+// Test allocation/reallocation using axom::allocate and axom::reallocate ints
+TEST(quest_initialize, int_reallocations)
+{
+  int* b3 = axom::allocate<int>(10);
+  EXPECT_NE(b3, nullptr);
+
+  b3 = axom::reallocate<int>(b3, 20);
+  EXPECT_NE(b3, nullptr);
+
+  axom::deallocate(b3);
+}
+
+#if defined AXOM_USE_SIDRE
+// Test allocation/reallocation using sidre::Buffer.
+TEST(quest_initialize, buffer_reallocations)
+{
+  axom::sidre::DataStore dataStore;
+
+  axom::sidre::Buffer* b1 =
+    dataStore.createBuffer(axom::sidre::DataTypeId::INT32_ID, 10);
+  b1->allocate();
+  EXPECT_NE(b1->getVoidPtr(), nullptr);
+
+  b1->reallocate(20);
+  EXPECT_NE(b1->getVoidPtr(), nullptr);
+
+  b1->deallocate();
+}
+
+// Test allocation/reallocation using sidre::View.
+TEST(quest_initialize, view_reallocations)
+{
+  axom::sidre::DataStore dataStore;
+  axom::sidre::Group* group = dataStore.getRoot();
+
+  axom::sidre::View* v3 = group->createView("v3", conduit::DataType::int32(10));
+  v3->allocate();
+  EXPECT_NE(v3->getVoidPtr(), nullptr);
+
+  v3->reallocate(20);
+  EXPECT_NE(v3->getVoidPtr(), nullptr);
+
+  v3->deallocate();
+}
+
+// Test immediately reserving space in UnstructuredMesh.
+TEST(quest_initialize, immediate_ug_reserve)
+{
+  axom::sidre::DataStore dataStore;
+  axom::sidre::Group* meshGroup = dataStore.getRoot()->createGroup("myGroup");
+  axom::mint::UnstructuredMesh<axom::mint::SINGLE_SHAPE> contourMesh(
+    2,
+    axom::mint::CellType::SEGMENT,
+    meshGroup);
+  contourMesh.reserveCells(10);  // This may unexpectedly crash.
+}
```

Comment:
I think this test needs to be guarded by `AXOM_MINT_USE_SIDRE`

Reply:
Yes.  Fixed.